File: python/6_kyu/bit_counting.py
# Bit Counting
# https://www.codewars.com/kata/526571aae218b8ee490006f4/train/python

def count_bits(n):
    bit_number = sum(list(map(int, list(str(bin(n)))[2:])))
    return bit_number
    
    
print(count_bits(1234))
print(count_bits(0))
print(count_bits(4))
print(count_bits(7))
print(count_bits(9))
print(count_bits(10))

File: javascript/6_kyu/bit_counting.js
// Bit Counting
// https://www.codewars.com/kata/526571aae218b8ee490006f4/train/javascript

function count(n) {
    var count = 0;
    for ( var i = 0; i < n.length; i++ ) {
        if ( n[i] == "1" ) {
            count++;
        }
    }
    return count;
}

function countBits(n) {
    var bit_number = n.toString(2).split("");
    return count(bit_number);
}

console.log(countBits(1234));
console.log(countBits(0));
console.log(countBits(4));
console.log(countBits(7));
console.log(countBits(9));
console.log(countBits(10));